tests/minimal_init.lua

```diff
@@ -0,0 +1,161 @@
+local M = {}
+
+---@alias PluginName string The plugin name, will be used as part of the git clone destination
+---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
+---@alias GitPlugins table<PluginName, PluginUrl> Plugins to clone with git
+
+-- Gets the current directory of this file
+local test_dir = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h")
+local rest_nvim_dir = vim.fn.fnamemodify(test_dir, ":h")
+
+---Gets the root directory of the minimal init and if path is specified appends the given path to the root allowing for
+---subdirectories within the current cwd
+---@param path string? The additional path to append to the root, not required
+---@return string root The root path suffixed with the path provided or an empty suffix if none was given
+function M.root(path)
+  return test_dir .. "/.deps/" .. (path or "")
+end
+
+---Returns the package root and ensures the path exists on disk
+---@return string pkg_root The package root suffixed with a `/`
+M.package_root = function()
+  local pkg_root = M.root("plugins/")
+  if not vim.uv.fs_stat(pkg_root) then
+    vim.fn.mkdir(pkg_root, "p")
+  end
+  return pkg_root
+end
+
+---Run a system command through `vim.system` and error on failed commands
+---@param cmd string[] The command to run
+---@param opts vim.SystemOpts? Options to pass to `vim.system`
+---@return vim.SystemCompleted
+function M.system(cmd, opts)
+  local out = vim.system(cmd, opts or {}):wait()
+  if out.code ~= 0 then
+    error(
+      string.format(
+        ">> Failed to run command: "
+          .. table.concat(cmd, " ")
+          .. "\n>> Exit code: %d\n>> Signal: %d\n===== STDOUT =====\n%s\n===== STDERR =====\n%s\n",
+        out.code,
+        out.signal,
+        out.stdout,
+        out.stderr
+      ),
+      vim.log.levels.ERROR
+    )
+  end
+  return out
+end
+
+---Make the `rest.nvim` lua rock and register the resulting plugin on the `runtimepath`
+function M.make_luarock()
+  print(">> Making rest.nvim luarock")
+  local luarocks_binary = "luarocks"
+  local install_destination = M.root("rocks/")
+  vim.g.rocks_nvim = {
+    rocks_path = install_destination,
+    luarocks_binary = luarocks_binary,
+  }
```

Comment:
Is this here as leftovers from a past changes iteration (or something around these lines) or do you plan to use `rocks.nvim` later for testing? 👀

Reply:
Nope, that's some leftovers 😅.

I went down the *horrid*, ***horrid***, road of not rolling my own and using other plugins to handle the luarocks stuff. They're all *way* too immature lacking basic stuff to just synchronously install a plugin.

I ended up rolling a super simple wrapper instead to call out to luarocks.

I'll remove it.

Reply:
I may end up PR'ing to `rocks.nvim` to fix the issues I ran into with it when it came to creating a test harness.

Reply:
Removed as part of the force push from `bbcf8ea` -> `45f9046`